src/index.js

```diff
@@ -42,11 +42,25 @@ async function main(config) {
 
     const transformed = await babelTransformFile(sourceFile, {
       root: config.root,
+      sourceMaps: config.sourceMaps,
     })
     await makeDir(path.dirname(outputFile))
-    await fs.writeFile(outputFile, transformed.code, {
-      mode: stats.mode,
-    })
+
+    const mapFile = outputFile.replace(/\..*$/, '.js.map')
```

Comment:
`outputFile` only ever has `.js` extension here, so you can get rid of the regexp replace